src/js/jsx/sections/style/EffectsPanel.jsx

```diff
@@ -176,7 +177,21 @@ define(function (require, exports, module) {
             dialog.toggle(event);
             this.getFlux().actions.layerEffects.addEffect(this.props.document, layers, effectType);
         },
-
+        
+        /**
+         * True if user can add new effect to the selected layers.
+         *
+         * @private
+         * @param {string} effectName - accept: strokeEffects, colorOverlays, innerShadows, dropShadows
+         * @return {boolean}
+         */
+        _canAddEffect: function (effectName) {
+            var effects = collection.pluck(this.props.document.layers.selected, effectName);
+            
+            // Return false if any of the selected layers reach maximum effect count.
+            return collection.zip(effects).size < MAX_EFFECT_COUNT;
```

Comment:
performance nit: `zip` is kind of a heavy function just for getting the max size.  It would be one thing if we were actually caching the value for use elsewhere, but here it is just wasted effort.  It may not really be significant in this case though.